nodemanager/scripts/EndTask.sh

```diff
@@ -14,7 +14,51 @@ taskFolder=$4
 
 isDockerTask=$(CheckDockerEnvFileExist $taskFolder)
 cgDisabled=$(CheckCgroupDisabledInFlagFile $taskFolder)
-if $CGInstalled && ! $cgDisabled; then
+if ! CGroupV1 && ! $cgDisabled; then
+	if $isDockerTask; then
+		containerId=$(GetContainerId $taskFolder)
+		groupName=$(GetCGroupNameOfDockerTaskV2 $containerId)
+	else
+		groupName=$(GetCGroupName "$taskId")
+	fi
+
+	if [ "$forced" == "1" ]; then
```

Comment:
So in cgroupv1, we need to manually kill the processes one by one by reading the pid, but this newly added file cgroup.kill in cgroupv2 can help us automatically kill all affected processes?

Reply:
Yes, though there's a limitation that cgroup v2 can only perform kill 9, but not graceful sigint. The reason why this is preferred is that linux kernel doc states that cgroup kill feature checks and ensures the correct order in which dependent processes are supposed to end.